drivers/entropy/entropy_mcux_caam.c

```diff
@@ -26,13 +28,26 @@ static int entropy_mcux_caam_get_entropy(const struct device *dev,
 	const struct mcux_entropy_config *config = dev->config;
 	status_t status;
 	caam_handle_t handle;
+	uint16_t read_length = 0;
+	uint16_t insert_idx = 0;
 
 	handle.jobRing = kCAAM_JobRing0;
 
-	status = CAAM_RNG_GetRandomData(
-			config->base, &handle, kCAAM_RngStateHandle0,
-			buffer, length, kCAAM_RngDataAny, NULL);
-	__ASSERT_NO_MSG(!status);
+	/*
+	 * The buffer passed to the CAAM RNG function needs to be in non-cache
+	 * memory. Use an intermediate buffer to stage the data to the user
+	 * buffer.
+	 */
+	while (insert_idx < length) {
```

Comment:
if we keep reading random number, the buffer will exhusted, do we need keep index here?